dsc/src/mcp/list_resources.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::mcp::McpServer;
+use dsc_lib::{
+    DscManager,
+    discovery::{
+        command_discovery::ImportedManifest,
+        discovery_trait::DiscoveryKind,
+    },
+    progress::ProgressFormat,
+};
+use rmcp::{ErrorData as McpError, Json, tool, tool_router};
+use schemars::JsonSchema;
+use serde::{Serialize, Deserialize};
+use tokio::task;
+
+#[derive(Serialize, Deserialize, JsonSchema)]
+pub struct ResourceListResult {
+    pub resources: Vec<ImportedManifest>,
+}
+
+#[tool_router]
+impl McpServer {
+    #[must_use]
+    pub fn new() -> Self {
+        Self {
+            tool_router: Self::tool_router()
+        }
+    }
+
+    #[tool(
+        description = "List all DSC resources available on the local machine",
+        annotations(
+            title = "Enumerate all available DSC resources on the local machine",
+            read_only_hint = true,
+            destructive_hint = false,
+            idempotent_hint = true,
+            open_world_hint = true,
+        )
+    )]
+    async fn list_resources(&self) -> Result<Json<ResourceListResult>, McpError> {
```

Comment:
It feels like a tool that uses the noun "resource" and shares a symbol with `ServerHandler` will be monumentally confusing...

I honestly don't know how the agent is going to handle the dual meaning of resource in the context of DSC+MCP.  But I do think it would be helpful for the humans reading this code for the tool to be named something else, even if it is:
```suggestion
    async fn list_dsc_resources(&self) -> Result<Json<ResourceListResult>, McpError> {
```

This might also be a better fit for MCP resources, possibly:
- A resource that is like `dsc resource list`; contains all resources with maybe `{ Type, Description }` to keep things short
- A resource template that is like `dsc resource list <Type> -o json` to have a more complete description.  This really only adds capabilities, but maybe more will go here eventually.

I think that the schema being part of the MCP resource template would be great, but I also think that the line between MCP resources and tools should be the need to invoke the DSC resource command or not.

Reply:
tools have descriptions, so I don't think the name of the tool is that critical?  I'm ok with changing the name, however.  For concerns about size/token constraints, my expectation is that the LLM would do a list which gives a short version of info and then a `get_resource` type tool to get full information